### PR TITLE
Better integration with wallhaven

### PR DESCRIPTION
**Pull request by artyomd**
Hey :wave: 

I have made a couple of changes regarding the wallhaven integration

- replaced the host url https://wallhaven.cc with the api url https://wallhaven.cc/api/v1/
- new field for setting your personal api key
- added support for fetching nsfw imaged (works only if the api key is set, otherwise wallhaven returns empty list)

### CODE REVIEW

File: randomwallpaper@iflow.space/settings.ui [hunk 1]
```diff
@@ -1004,6 +1019,13 @@ You can also define a prefix that will be added to the image URL.</property>
             <property name="can-focus">1</property>
           </object>
         </child>
+        <child>
+          <object class="GtkCheckButton" id="wallhaven-allow-nsfw">
+            <property name="label" translatable="yes">NSFW</property>
```
Comment (ifl0w) [inline randomwallpaper@iflow.space/settings.ui:1024]: I think, the information that an API key is required would be a good addition to the name (i.e., "NSFW (API key required)")

File: randomwallpaper@iflow.space/sourceAdapter.js [hunk 2]
```diff
@@ -204,50 +205,32 @@ var WallhavenAdapter = class extends BaseAdapter {
...
+			if(apiKey){
+                imageUrl += "?apikey="+apiKey;
+            }
+
+            if (callback) {
+                let historyEntry = new HistoryModule.HistoryEntry(null, 'Wallhaven', imageUrl);
+                historyEntry.source.sourceUrl = 'https://wallhaven.cc/';
+                historyEntry.source.imageLinkUrl = url;
+                callback(historyEntry);
+            }
```
Comment (ifl0w) [inline randomwallpaper@iflow.space/sourceAdapter.js:233]: The indentation here is a little bit off.

**Review by ifl0w — COMMENTED** [review]
Wow, great work! I did not know that they have an API now! 
There are only two minor "issues" but I'll definitely merge these changes once the issues are resolved! I can do that myself the next time I am working on this project if you do not find the time!

Thanks for the time and effort!

File: randomwallpaper@iflow.space/sourceAdapter.js (hunk 2 above)
Comment (artyomd) [reply randomwallpaper@iflow.space/sourceAdapter.js:233]: honesty, I don't see any indentation mistake in here. could you please explain how you want this to be changed?

Comment (ifl0w) [reply randomwallpaper@iflow.space/sourceAdapter.js:233]: You used spaces for the indentation while the rest of the file uses tabs. But that is not really a problem for me. I'll simply reformat the code on the target branch. Thanks!

**Pull request merged**